### ISSUE TITLE
[Morphy] VPC full refresh fails with 'You must wait before you can make ghost-tags get api requests with attached_to query parameter'

### ISSUE THREAD

**Issue by nasark** [ManageIQ/manageiq-providers-ibm_cloud#285]
This was encountered on a downstream cluster, however upon re-running the refresh was successful.

Steps:
1. Create an IBM Cloud VPC provider
2. Initiate a full refresh

```
{"@timestamp":"2021-09-24T15:26:05.398572 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-4-7cd7cdd5f7-fqjcc","pid":7,"tid":"8015337940","service":"evm","level":"err","message":"MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::Refresher#refresh) EMS: [VPC], id: [4] Refresh failed"}
{"@timestamp":"2021-09-24T15:26:05.399583 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-4-7cd7cdd5f7-fqjcc","pid":7,"tid":"8015337940","service":"evm","level":"err","message":"[IBMCloudSdkCore::ApiException]: Error: , Code: , Information: You must wait 16991 ms before you can make ghost-tags get api requests with attached_to query parameter, X-global-transaction-id: gst-889cab30-e4eb-4902-8a78-f03db105d2af  Method:[block (2 levels) in <class:LogProxy>]"}
{"@timestamp":"2021-09-24T15:26:05.399975 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-4-7cd7cdd5f7-fqjcc","pid":7,"tid":"8015337940","service":"evm","level":"err","message":"/opt/IBM/infrastructure-management-gemset/gems/ibm_cloud_sdk_core-1.1.3/lib/ibm_cloud_sdk_core/base_service.rb:113:in `request'\n/opt/IBM/infrastructure-management-gemset/gems/ibm_cloud_global_tagging-0.1.1/lib/ibm_cloud_global_tagging/global_tagging_v1.rb:116:in `list_tags'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/sdk/branch.rb:80:in `send_request'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/sdk/branch.rb:51:in `request'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb:33:in `block in each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb:32:in `loop'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb:32:in `each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:88:in `each'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:88:in `to_a'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:88:in `tags_by_crn'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:64:in `block in instances'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:47:in `block (2 levels) in each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:47:in `each'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:47:in `block in each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:40:in `loop'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:40:in `each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:44:in `each'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:44:in `instances'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:18:in `parse'\n/var/www/miq/vmdb/app/models/manageiq/providers/inventory.rb:42:in `block in parse'\n/var/www/miq/vmdb/app/models/manageiq/providers/inventory.rb:39:in `each'\n/var/www/miq/vmdb/app/models/manageiq/providers/inventory.rb:39:in `parse'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:125:in `parse_targeted_inventory'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:92:in `block in refresh_targets_for_ems'\n/opt/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:20:in `realtime_store'\n/opt/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:56:in `realtime_block'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:91:in `refresh_targets_for_ems'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:41:in `block (2 levels) in refresh'\n/opt/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:20:in `realtime_store'\n/opt/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:62:in `realtime_block'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:41:in `block in refresh'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:31:in `each'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:31:in `refresh'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:11:in `refresh'\n/var/www/miq/vmdb/app/models/ems_refresh.rb:71:in `block in refresh'\n/var/www/miq/vmdb/app/models/ems_refresh.rb:70:in `each'\n/var/www/miq/vmdb/app/models/ems_refresh.rb:70:in `refresh'\n/var/www/miq/vmdb/app/models/miq_queue.rb:484:in `block in dispatch_method'\n/usr/share/ruby/timeout.rb:93:in `block in timeout'\n/usr/share/ruby/timeout.rb:33:in `block in catch'\n/usr/share/ruby/timeout.rb:33:in `catch'\n/usr/share/ruby/timeout.rb:33:in `catch'\n/usr/share/ruby/timeout.rb:108:in `timeout'\n/var/www/miq/vmdb/app/models/miq_queue.rb:482:in `dispatch_method'\n/var/www/miq/vmdb/app/models/miq_queue.rb:459:in `block in deliver'\n/var/www/miq/vmdb/app/models/user.rb:376:in `with_user_group'\n/var/www/miq/vmdb/app/models/miq_queue.rb:459:in `deliver'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:80:in `deliver_queue_message'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:104:in `deliver_message'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:122:in `block in do_work'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:118:in `loop'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:118:in `do_work'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:267:in `block in do_work_loop'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:264:in `loop'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:264:in `do_work_loop'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:114:in `run'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:96:in `start'\n/var/www/miq/vmdb/lib/workers/bin/run_single_worker.rb:128:in `<main>'"}
{"@timestamp":"2021-09-24T15:26:05.400283 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-4-7cd7cdd5f7-fqjcc","pid":7,"tid":"8015337940","service":"evm","level":"err","message":"MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::Refresher#refresh) EMS: [VPC], id: [4] Unable to perform refresh for the following targets:"}
{"@timestamp":"2021-09-24T15:26:05.401206 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-4-7cd7cdd5f7-fqjcc","pid":7,"tid":"8015337940","service":"evm","level":"err","message":"MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::Refresher#refresh)  --- ManageIQ::Providers::IbmCloud::VPC::CloudManager [VPC] id [4]"}
```

**Comment by nasark** [comment]
@miq-bot add-labels bug

**Comment by Fryguy** [comment]
Are we getting throttled?

**Comment by nasark** [comment]
@Fryguy I'm not too sure about that but just to add on to this issue, a similar tag-related timeout was seen but with a different message: `Timed out after using the allocated 60 seconds `

```
"@timestamp":"2021-09-24T16:15:57.162318 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-9-5849568d77-p5lfw","pid":7,"tid":"2ab66de31964","service":"evm","level":"info","message":"MIQ(ManageIQ::Providers::Ib
mCloud::VPC::CloudManager::Refresher#refresh) EMS: [nasar's VPC], id: [9]   ManageIQ::Providers::IbmCloud::VPC::CloudManager [nasar's VPC] id [9]"}
{"@timestamp":"2021-09-24T16:15:57.328380 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-9-5849568d77-p5lfw","pid":7,"tid":"2ab66de31964","service":"evm","level":"info","message":"MIQ(ManageIQ::Providers::Ib
mCloud::VPC::CloudManager::Refresher#refresh_targets_for_ems) EMS: [nasar's VPC], id: [9] Refreshing target ManageIQ::Providers::IbmCloud::VPC::CloudManager [nasar's VPC] id [9]..."}
{"@timestamp":"2021-09-24T16:15:57.328664 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-9-5849568d77-p5lfw","pid":7,"tid":"2ab66de31964","service":"evm","level":"debug","message":"MIQ(ManageIQ::Providers::I
bmCloud::VPC::CloudManager::Refresher#parse_targeted_inventory) EMS: [nasar's VPC], id: [9] Parsing inventory..."}
{"@timestamp":"2021-09-24T16:18:09.498609 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-9-5849568d77-p5lfw","pid":7,"tid":"2ab66de31964","service":"evm","level":"err","message":"MIQ(ManageIQ::Providers::Ibm
Cloud::VPC::CloudManager::Refresher#refresh) EMS: [nasar's VPC], id: [9] Refresh failed"}
{"@timestamp":"2021-09-24T16:18:09.498889 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-9-5849568d77-p5lfw","pid":7,"tid":"2ab66de31964","service":"evm","level":"err","message":"[HTTP::TimeoutError]: Timed 
out after using the allocated 60 seconds  Method:[block (2 levels) in <class:LogProxy>]"}
{"@timestamp":"2021-09-24T16:18:09.499004 ","hostname":"1-ibm-cloud-vpc-cloud-refresh-9-5849568d77-p5lfw","pid":7,"tid":"2ab66de31964","service":"evm","level":"err","message":"/opt/IBM/infrastructure-mana
gement-gemset/gems/http-4.4.1/lib/http/timeout/global.rb:125:in `log_time'\n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/timeout/global.rb:107:in `wait_readable_or_timeout'\n/opt/IBM
/infrastructure-management-gemset/gems/http-4.4.1/lib/http/timeout/global.rb:89:in `block in perform_io'\n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/timeout/global.rb:84:in `loop'\
n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/timeout/global.rb:84:in `perform_io'\n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/timeout/global.rb:50:in `readpa
rtial'\n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/connection.rb:212:in `read_more'\n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/connection.rb:103:in `read_h
eaders!'\n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/client.rb:75:in `perform'\n/opt/IBM/infrastructure-management-gemset/gems/http-4.4.1/lib/http/client.rb:31:in `request'\n/opt/I
BM/infrastructure-management-gemset/gems/ibm_cloud_sdk_core-1.1.3/lib/ibm_cloud_sdk_core/base_service.rb:103:in `request'\n/opt/IBM/infrastructure-management-gemset/gems/ibm_cloud_global_tagging-0.1.1/lib
/ibm_cloud_global_tagging/global_tagging_v1.rb:116:in `list_tags'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_too
ls/sdk/branch.rb:80:in `send_request'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/sdk/branch.rb:51:in `requ
est'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb:33:in `block in each_resource'\n/opt/IBM/infra
structure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb:32:in `loop'\n/opt/IBM/infrastructure-management-gemset/bundler/ge
ms/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb:32:in `each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud
-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:88:in `each'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/man
ageiq/providers/ibm_cloud/inventory/collector/vpc.rb:88:in `to_a'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/in
ventory/collector/vpc.rb:88:in `tags_by_crn'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb
:64:in `block in instances'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:47:in `block (2 levels) in e
ach_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:47:in `each'\n/opt/IBM/infrastructure-mana
gement-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:47:in `block in each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gem
s/bluecf-providers-ibm_cloud-d826bf930c74/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:40:in `loop'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/l
ib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb:40:in `each_resource'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_c
loud/inventory/parser/vpc.rb:44:in `each'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:44
:in `instances'\n/opt/IBM/infrastructure-management-gemset/bundler/gems/bluecf-providers-ibm_cloud-d826bf930c74/app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:18:in `parse'\n/var/www/miq/
vmdb/app/models/manageiq/providers/inventory.rb:42:in `block in parse'\n/var/www/miq/vmdb/app/models/manageiq/providers/inventory.rb:39:in `each'\n/var/www/miq/vmdb/app/models/manageiq/providers/inventory
.rb:39:in `parse'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:125:in `parse_targeted_inventory'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:92
:in `block in refresh_targets_for_ems'\n/opt/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:20:in `realtime_store'\n/opt
/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:56:in `realtime_block'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:91:in `refresh_targets_for_ems'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:41:in `block (2 levels) in refresh'\n/opt/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:20:in `realtime_store'\n/opt/IBM/infrastructure-management-gemset/gems/more_core_extensions-4.3.1/lib/more_core_extensions/core_ext/benchmark/realtime_store.rb:62:in `realtime_block'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:41:in `block in refresh'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:31:in `each'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:31:in `refresh'\n/var/www/miq/vmdb/app/models/manageiq/providers/base_manager/refresher.rb:11:in `refresh'\n/var/www/miq/vmdb/app/models/ems_refresh.rb:71:in `block in refresh'\n/var/www/miq/vmdb/app/models/ems_refresh.rb:70:in `each'\n/var/www/miq/vmdb/app/models/ems_refresh.rb:70:in `refresh'\n/var/www/miq/vmdb/app/models/miq_queue.rb:484:in `block in dispatch_method'\n/usr/share/ruby/timeout.rb:93:in `block in timeout'\n/usr/share/ruby/timeout.rb:33:in `block in catch'\n/usr/share/ruby/timeout.rb:33:in `catch'\n/usr/share/ruby/timeout.rb:33:in `catch'\n/usr/share/ruby/timeout.rb:108:in `timeout'\n/var/www/miq/vmdb/app/models/miq_queue.rb:482:in `dispatch_method'\n/var/www/miq/vmdb/app/models/miq_queue.rb:459:in `block in deliver'\n/var/www/miq/vmdb/app/models/user.rb:376:in `with_user_group'\n/var/www/miq/vmdb/app/models/miq_queue.rb:459:in `deliver'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:80:in `deliver_queue_message'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:104:in `deliver_message'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:122:in `block in do_work'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:118:in `loop'\n/var/www/miq/vmdb/app/models/miq_queue_worker_base/runner.rb:118:in `do_work'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:267:in `block in do_work_loop'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:264:in `loop'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:264:in `do_work_loop'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:114:in `run'\n/var/www/miq/vmdb/app/models/miq_worker/runner.rb:96:in `start'\n/var/www/miq/vmdb/lib/workers/bin/run_single_worker.rb:128:in `<main>'"}
```

**Comment by nasark** [comment]
@Fryguy @agrare I am unable to reproduce this, I suspect it may have been due to intermittent network issues on the IBM Cloud side. Can we close this issue?

**Comment by agrare** [comment]
We hit this again, since we are requesting attached tags per-instance with a high number of instances it is more likely to occur.  I think we should catch this exception and sleep for a bit before retrying.

**Comment by nasark** [comment]
https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/441

**Comment by Fryguy** [comment]
Closed via #441
